### ISSUE TITLE
don't panic in middleware.

### ISSUE THREAD

**Issue by james-lawrence** [supertokens/supertokens-golang#341]
calling supertokens middleware can result in panics if init wasn't called (ran into this in our test suite). better to return an error or just configure the middleware to log and return a status code for all requests. potentially nuking the entire program isn't ideal.

github.com/supertokens/supertokens-golang/supertokens/main.go:37
```Please call supertokens.Init function before using the Middleware```

**Comment by rishabhpoddar** [comment]
This change will require a lot of resources on our end. Let's wait to see more traction for this change.

**Comment by rishabhpoddar** [comment]
Thanks to @james-lawrence, a good middleground to change our sdk is:

```go
func Middleware(theirHandler http.Handler) http.Handler {
    instance, err := GetInstanceOrThrowError()
    if err != nil {
        return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
            http.Error(w, "misconfigured auth handler", http.StatusInternalServerError)
        })
    }
    return instance.middleware(theirHandler)
}
```

Instead of panic, we return a 500 to the client.

**Comment by rishabhpoddar** [comment]
This has been released in golang sdk version 0.15.0
